apps/earn-protocol/hooks/use-sdk.ts

```diff
@@ -31,25 +32,111 @@ export const useSDK = () => {
     [sdk],
   )
 
-  const getFleetClient = useMemo(
+  const getNewDepositTX = useMemo(
     () =>
-      async ({ chainId, fleetAddress }: { chainId: number; fleetAddress: string }) => {
+      async ({
+        chainId,
+        fleetAddress,
+        walletAddress,
+        amount,
+      }: {
+        chainId: number
+        fleetAddress: string
+        walletAddress: string
+        amount: string
+      }) => {
         const chain = await sdk.chains.getChainById({
           chainId,
         })
 
         if (!chain) {
           throw new Error('Chain not found')
         }
-        const fleet = chain.armada.getFleet({
-          address: Address.createFromEthereum({ value: fleetAddress }),
+        const { chainInfo } = chain
+
+        const poolId = ArmadaPoolId.createFrom({
+          chainInfo,
+          fleetAddress: Address.createFromEthereum({ value: fleetAddress }),
+          protocol: ArmadaProtocol.createFrom({ chainInfo }),
+        })
+
+        const poolInfo = await sdk.armada.getPoolInfo({
+          poolId,
+        })
+
+        const user = User.createFrom({
+          chainInfo,
+          wallet: Wallet.createFrom({
+            address: Address.createFromEthereum({ value: walletAddress }),
+          }),
+        })
+
+        const tokenAmount = TokenAmount.createFrom({
+          amount,
+          token: poolInfo.totalDeposits.token,
+        })
+
+        return sdk.armada.getNewDepositTX({
+          poolId,
+          user,
+          amount: tokenAmount,
+        })
+      },
+    [sdk],
+  )
+
+  const getWithdrawTX = useMemo(
+    () =>
+      async ({
+        chainId,
+        fleetAddress,
+        walletAddress,
+        amount,
+      }: {
+        chainId: number
+        fleetAddress: string
+        walletAddress: string
+        amount: string
+      }) => {
+        const chain = await sdk.chains.getChainById({
+          chainId,
         })
 
-        if (!fleet) {
-          throw new Error(`SDK: Fleet not found: ${fleetAddress}`)
+        if (!chain) {
+          throw new Error('Chain not found')
         }
+        const { chainInfo } = chain
+
+        const poolId = ArmadaPoolId.createFrom({
+          chainInfo,
+          fleetAddress: Address.createFromEthereum({ value: fleetAddress }),
+          protocol: ArmadaProtocol.createFrom({ chainInfo }),
+        })
 
-        return fleet
+        const poolInfo = await sdk.armada.getPoolInfo({
```

Comment:
not sure about the getPoolInfo and other metohds calling rpc - do we need to check if there is any return data, or that's handeled elswhere and it throws ?

Reply:
Yes, it is handled in the server and an exception is thrown if the pool cannot be found